### DISCUSSION TITLE
[Q] SpliceColumn does not insert "0"

### DISCUSSION THREAD

**Discussion by sshim44** [exceljs/exceljs#2361] [Q&A]
## 💬 Questions and Help

Please do not use issues for making a questions, for this purposes much better is GitHub Discussions feature:
https://github.com/exceljs/exceljs/discussions/categories/q-a

### How to make a question?

Hello,

It seems like the "spliceColumns" function does not insert 0 values.

I am trying to insert values below into a column, but it seems like it only inserts if the value is either greater or less than 0.

<img width="326" alt="image" src="https://github.com/exceljs/exceljs/assets/134318091/59f2dace-7f99-40f8-9e01-76ea6577036d">

Below is the result from the SpliceColumns function.

<img width="99" alt="image" src="https://github.com/exceljs/exceljs/assets/134318091/c7471734-b4a8-4e89-bb82-432d987f10bc">

Why does the function not insert 0? Is that expected? If so, is there a workaround?

Please let me know if I can provide additional information.

Thanks,
Simon

**Comment by wedsonxse** [comment]
The spliceColumns methods does a verification `Value || null` for valid values, since 0 is treated as a false in JS boolean operations, you column will end up with some blank cells, because it will be changed to null.

What you can do is simply iterate over your sheet cells after using the spliceColumns(), changing your `null` values to 0. Even being not an optimized solution, will solve your problem.

Considering the way JS handle some false values in operations, maybe the simple 'Value || null' verification should be better analyzed, but to keep consistency and atomicity with the language, the current implementation works fine even requiring some post operations for final result tuning.
